db/src/main/kotlin/edu/artic/db/daos/ArticMapAnnotationDao.kt

```diff
@@ -16,17 +17,24 @@ abstract class ArticMapAnnotationDao {
     @Query("select * from ArticMapAnnotation where annotationType = :type")
     abstract fun getAnnotationByType(type: String): Flowable<List<ArticMapAnnotation>>
 
-    @Query("select * from ArticMapAnnotation where annotationType = \"Text\" and textType = :type")
+    @Query("select * from ArticMapAnnotation where annotationType = :type and floor = :floor")
+    abstract fun getAnnotationByTypeForFloor(type: String, floor: String): Flowable<List<ArticMapAnnotation>>
+
+    @Query("select * from ArticMapAnnotation where annotationType = \"Text\" and textType = :type ")
```

Comment:
for `Text` can you reuse `annotationType` like so:
```
 @Query("select * from ArticMapAnnotation where annotationType = :annotationType and textType = :type ")
   abstract fun getTextAnnotationByType(type: String, annotationType: String = ArticMapAnnotationType.TEXT): Flowable<List<ArticMapAnnotation>>
```

Reply:
So use this if we want default?